### DISCUSSION TITLE
Custom Color Mappings

### DISCUSSION THREAD

**Discussion by acedrew** [marcusolsson/grafana-hourly-heatmap-panel#56] [Ideas]
Many combinations of colors in the custom mapping result in null color values and nothing rendered.
I found basically any combination moving from blue to red through purple resulted in a failed mapping. 
I would like to contribute to a more generic gradient designer for grafana, that could be used across panels. I've dug pretty deep into this for some of my plugins, but it seems silly reinvent the wheel each time.

**Comment by marcusolsson** [comment]
Thank you for reporting this! I'll try and see if I can reproduce this. 

A generic gradient designer would be incredibly useful. If you end up publishing one, I'd be happy to incorporate it.

**Comment by acedrew** [comment]
I'll try to get some feedback from grafana to see what the best way to work
on this would be.

**Comment by marcusolsson** [comment]
Now that https://github.com/grafana/grafana/pull/28039 was merged, I hope to add this for the next version. Won't work with older versions though, so may look at keeping the current editor as a fallback.

**Comment by marcusolsson** [comment]
I've added support for the new Color scheme field option introduced in Grafana 7.3: https://github.com/marcusolsson/grafana-hourly-heatmap-panel/commit/c4dce84d9bfd6c0216e588ea9ed1b855969b41cb

Unfortunately, it doesn't yet have support for creating custom color schemes. I'll see if I can improve the current custom palette editor.

**Comment by stevemandl** [comment]
The custom color palette with negative values seems to break too. All values less than zero display the same color rather than making a smooth gradient.
![Screenshot from 2021-11-02 09-09-42](https://user-images.githubusercontent.com/5465626/139853438-811c3109-2a30-45f5-9660-96f0a48cc752.png)

**Comment by kolaente** [comment]
My heatmap looks like this:

![image](https://user-images.githubusercontent.com/13721712/149789597-cba76c5d-836c-4e90-964d-86dd246a8062.png)

Did I misconfigure something or is this currently broken?
